### PR TITLE
Bugfix: Group key map validation fixed.

### LINKED ISSUE #22236: [SVE2][TC-G-2.2]Group Key map
In TC-G-2.2:

In the pre-requisite steps to the TC
- `ubuntu@ubuntu:~/apps$ ./chip-tool groupkeymanagement write group-key-map '[{"groupId": 1, "groupKeySetID": 1, "fabricIndex": 1},{"groupId": 2, "groupKeySetID": 1, "fabricIndex": 1},{"groupId": 3, "groupKeySetID": 1, "fabricIndex": 1}]' 1 0` will fail, even though the spec says

> The length of the <<ref_GroupKeyMap>> and <<ref_GroupTable>> list attributes SHALL NOT exceed the value of the `MaxGroupsPerFabric` attribute multiplied by the number of <<ref_SupportedFabricsAttribute,supported fabrics>>.

The code has:

    GroupDataProvider(uint16_t maxGroupsPerFabric    = CHIP_CONFIG_MAX_GROUPS_PER_FABRIC,
                      uint16_t maxGroupKeysPerFabric = CHIP_CONFIG_MAX_GROUP_KEYS_PER_FABRIC) :
        mMaxGroupsPerFabric(maxGroupsPerFabric),
        mMaxGroupKeysPerFabric(maxGroupKeysPerFabric)
    {}

There, `CHIP_CONFIG_MAX_GROUP_KEYS_PER_FABRIC = 2`

The check has:

    // Insert last
    VerifyOrReturnError(fabric.map_count == index, CHIP_ERROR_INVALID_ARGUMENT);
    VerifyOrReturnError(fabric.map_count < mMaxGroupKeysPerFabric, CHIP_ERROR_INVALID_LIST_LENGTH);

 #### Proposed Solution
    VerifyOrReturnError(fabric.map_count < mMaxGroupsPerFabric, CHIP_ERROR_INVALID_LIST_LENGTH);

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
#### Problem
* Fixes [#22236: Group Key map length validation.](https://github.com/project-chip/connectedhomeip/issues/22236)

#### Change overview
* Validation fixed.
* Tests updated.

#### Testing
* Group and GroupKeyManagement YAML tests run successfully.

### CODE REVIEW

**Review by tcarmelveilleux — APPROVED** [review]

**Comment by andy31415** [comment]
Accepted spec bugfix in master.

File: src/credentials/GroupDataProviderImpl.cpp [hunk 1]
```diff
@@ -1420,7 +1421,7 @@ CHIP_ERROR GroupDataProviderImpl::SetGroupKeyAt(chip::FabricIndex fabric_index,
 
     // Insert last
     VerifyOrReturnError(fabric.map_count == index, CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(fabric.map_count < mMaxGroupKeysPerFabric, CHIP_ERROR_INVALID_LIST_LENGTH);
+    VerifyOrReturnError(fabric.map_count < mMaxGroupsPerFabric, CHIP_ERROR_INVALID_LIST_LENGTH);
```
Comment (bzbarsky-apple) [inline src/credentials/GroupDataProviderImpl.cpp:1424]: Do we use mMaxGroupKeysPerFabric for anything after this?  Where does it restrict the number of keysets we can add?

Comment (tcarmelveilleux) [reply src/credentials/GroupDataProviderImpl.cpp:1424]:
```
CHIP_ERROR GroupDataProviderImpl::SetKeySet(chip::FabricIndex fabric_index, const ByteSpan & compressed_fabric_id,
                                            const KeySet & in_keyset)
```

This is what should use mMaxGroupKeysPerFabric, but it doesn't. We can currently add infinite group key sets.

We update the count but never check we reach a max:
```
   // Update fabric
    fabric.keyset_count++;
    fabric.first_keyset = in_keyset.keyset_id;
    return fabric.Save(mStorage);
```

Comment (rcasallas-silabs) [reply src/credentials/GroupDataProviderImpl.cpp:1424]: You're correct, the MaxGroupKeysPerFabric limit was not enforced. I have added the missing validation, and updated the test accordingly.

**Review by jepenven-silabs — APPROVED** [review]

**Pull request merged**